src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -460,516 +886,228 @@ private static int GetFrequency(double value)
-            {
-                _ldaTrainer.Dispose();
-            }
+        protected override IRowMapper MakeRowMapper(ISchema schema)
+        {
+            return new Mapper(this, Schema.Create(schema));
         }
+    }
 
-        private ColumnType[] InitColumnTypes(int numTopics)
+    /// <include file='doc.xml' path='doc/members/member[@name="LightLDA"]/*' />
+    public sealed class LdaEstimator : IEstimator<LdaTransform>
+    {
+        internal static class Defaults
         {
-            Host.Assert(Utils.Size(Infos) > 0);
-            var types = new ColumnType[Infos.Length];
-            for (int c = 0; c < Infos.Length; c++)
-                types[c] = new VectorType(NumberType.Float, numTopics);
-            return types;
+            public const int NumTopic = 100;
+            public const Single AlphaSum = 100;
+            public const Single Beta = 0.01f;
+            public const int Mhstep = 4;
+            public const int NumIterations = 200;
+            public const int LikelihoodInterval = 5;
+            public const int NumThreads = 0;
+            public const int NumMaxDocToken = 512;
+            public const int NumSummaryTermPerTopic = 10;
+            public const int NumBurninIterations = 10;
+            public const bool ResetRandomGenerator = false;
         }
 
-        protected override ColumnType GetColumnTypeCore(int iinfo)
+        private readonly IHost _host;
+        private readonly LdaTransform.ColumnInfo[] _columns;
+
+        /// <include file='doc.xml' path='doc/members/member[@name="LightLDA"]/*' />
+        /// <param name="env">The environment.</param>
+        /// <param name="inputColumn">The column containing text to tokenize.</param>
+        /// <param name="outputColumn">The column containing output tokens. Null means <paramref name="inputColumn"/> is replaced.</param>
+        /// <param name="numTopic">The number of topics in the LDA.</param>
+        /// <param name="alphaSum">Dirichlet prior on document-topic vectors.</param>
+        /// <param name="beta">Dirichlet prior on vocab-topic vectors.</param>
+        /// <param name="mhstep">Number of Metropolis Hasting step.</param>
+        /// <param name="numIterations">Number of iterations.</param>
+        /// <param name="likelihoodInterval">Compute log likelihood over local dataset on this iteration interval.</param>
+        /// <param name="numThreads">The number of training threads. Default value depends on number of logical processors.</param>
+        /// <param name="numMaxDocToken">The threshold of maximum count of tokens per doc.</param>
+        /// <param name="numSummaryTermPerTopic">The number of words to summarize the topic.</param>
+        /// <param name="numBurninIterations">The number of burn-in iterations.</param>
+        /// <param name="resetRandomGenerator">Reset the random number generator for each document.</param>
+        public LdaEstimator(IHostEnvironment env,
```

Comment:
>LdaEstimator [](start = 15, length = 12)

`LatentDirichletAllocationEstimator`

Reply:
Keeping it LdaEstimator for now.  Please see my response in other comment where I rename  LdaTransformer

---
In reply to: [232437894](https://github.com/dotnet/machinelearning/pull/1410#discussion_r232437894) [](ancestors = 232437894)